### ISSUE TITLE
admin settings form should inherit from ConfigFormBase

### ISSUE THREAD

**Issue by joachim-n** [drupal-code-builder/drupal-code-builder#137]

**Comment by joachim-n** [comment]
And then needs a scaffold method getEditableConfigNames

**Comment by joachim-n** [comment]
And then some sample code in submitForm() to save the config.

And also sample of loading config in buildForm().

**Comment by joachim-n** [comment]
```
   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
   *   The factory for configuration objects.

    parent::__construct($config_factory);

      $container->get('config.factory'),


  public function buildForm(array $form, FormStateInterface $form_state) {
    $config = $this->config('MYMODULE.settings');

      // ...
      '#default_value' => $config->get('journey_invalid_page'),


  /**
   * {@inheritdoc}
   */
  public function submitForm(array &$form, FormStateInterface $form_state) {
    parent::submitForm($form, $form_state);
    $config = $this->config('MYMODULE.settings');

    if ($form_state->hasValue('TODO')) {
      $config->set('TODO', $form_state->getValue('TODO'));
    }

    $config->save();
  }


  /**
   * {@inheritdoc}
   */
  protected function getEditableConfigNames() {
    return ['MYMODULE.settings'];
  }



```

**Comment by joachim-n** [comment]
Closed by 2072b207c98ffcddcd8ed669f367094ab5023cb7 and subsequent.
